From ltsi-dev-bounces@lists.linuxfoundation.org Wed Oct 29 09:53:51 2014
From: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
Date: Wed, 29 Oct 2014 08:52:55 +0800
Subject: [LTSI-dev] [PATCH 5/8] pwm: lpss: Add ACPI and PCI IDs for Intel Braswell
To: LTSI Mailing List <ltsi-dev@lists.linuxfoundation.org>
Cc: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
Message-ID: <1414543978-24145-6-git-send-email-rebecca.swee.fun.chang@intel.com>


From: Alan Cox <alan@linux.intel.com>

This is pretty much the same as Baytrail PWM. Only difference is that the
input clock runs on different frequency.

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Mika Westerberg <mika.westerberg@linux.intel.com>
Signed-off-by: Thierry Reding <thierry.reding@gmail.com>
(cherry picked from commit 373c57829a3f9da1405b1fbd3d17e50f8e1f476e)

Signed-off-by: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
---
 drivers/pwm/pwm-lpss.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/pwm/pwm-lpss.c
+++ b/drivers/pwm/pwm-lpss.c
@@ -48,6 +48,11 @@ static const struct pwm_lpss_boardinfo b
 	25000000
 };
 
+/* Braswell */
+static const struct pwm_lpss_boardinfo bsw_info = {
+	19200000
+};
+
 static inline struct pwm_lpss_chip *to_lpwm(struct pwm_chip *chip)
 {
 	return container_of(chip, struct pwm_lpss_chip, chip);
@@ -189,6 +194,8 @@ static void pwm_lpss_remove_pci(struct p
 static struct pci_device_id pwm_lpss_pci_ids[] = {
 	{ PCI_VDEVICE(INTEL, 0x0f08), (unsigned long)&byt_info},
 	{ PCI_VDEVICE(INTEL, 0x0f09), (unsigned long)&byt_info},
+	{ PCI_VDEVICE(INTEL, 0x2288), (unsigned long)&bsw_info},
+	{ PCI_VDEVICE(INTEL, 0x2289), (unsigned long)&bsw_info},
 	{ },
 };
 MODULE_DEVICE_TABLE(pci, pwm_lpss_pci_ids);
@@ -231,6 +238,7 @@ static int pwm_lpss_remove_platform(stru
 
 static const struct acpi_device_id pwm_lpss_acpi_match[] = {
 	{ "80860F09", (unsigned long)&byt_info },
+	{ "80862288", (unsigned long)&bsw_info },
 	{ },
 };
 MODULE_DEVICE_TABLE(acpi, pwm_lpss_acpi_match);
